fs/osfs.py

```
@@ -440,6 +440,9 @@ def copy(self, src_path, dst_path, overwrite=False, preserve_time=False):
                     self.getsyspath(_src_path),
                     self.getsyspath(_dst_path),
                 )
+                # exit early if we copy the file onto itself
+                if overwrite and _src_sys == _dst_sys:
+                    return
```

Should we raise an Exception, rather than just silently returning?